vscode-dotnet-runtime-library/src/Utils/TypescriptUtilities.ts

```diff
@@ -157,6 +157,71 @@ export async function executeWithLock<A extends any[], R>(eventStream: IEventStr
     return returnResult;
 }
 
+const possiblyUsefulUpperCaseEnvVars = new Set<string>([ // This is a local variable instead of in the function so it doesn't get recreated every time the function is called. I looked at compiled JS and didn't see it get optimized.
+    'COMMONPROGRAMFILES',
+    'COMMONPROGRAMFILES(x86)',
+    'PATH',
+    'SYSTEMROOT',
+    'PROGRAMFILES',
+    'POWERSHELL_DISTRIBUTION_CHANNEL',
+    'PROCESSOR_IDENTIFIER',
+    'PSMODULEPATH',
+    'PROCESSOR_ARCHITECTURE',
+    'VSCODE_CLI',
+    'VSCODE_CODE_CACHE_PATH',
+    'VSCODE_HANDLES_UNCAUGHT_ERRORS',
+    'RESOLVEDLANGUAGE',
+    'VSCODE_PID',
+    'WINDIR',
+    'DIRCMD',
+    'TERM_PROGRAM_VERSION',
+    'ALLUSERSPROFILE',
+    'COMSPEC',
+    'DOTNET_MULTILEVEL_LOOKUP',
+    'ELECTRON_RUN_AS_NODE',
+    'LANG',
+    'HOME',
+    'PATHEXT',
+    'SHELL',
+    'TERM',
+    'PWD',

```

Comment:
Or this. I generally think that if it isn't specific to us, it might be safer to not grab it

Reply:
The shell/term has a lot of influential factors on whether our code works or not, so I don't want to remove that, but I see your point.